sentinelhub/aws_safe.py

```diff
@@ -195,75 +213,88 @@ def get_safe_struct(self):
         :return: nested dictionaries representing .SAFE structure
         :rtype: dict
         """
-        # pylint: disable=too-many-branches
-        safe = {}
-        main_folder = self.get_main_folder()
-        safe[main_folder] = {}
+        return {
+            self.get_main_folder(): {
+                AwsConstants.AUX_DATA: self._get_aux_substruct(),
+                AwsConstants.IMG_DATA: self._get_imamge_substruct(),
+                AwsConstants.QI_DATA: self._get_qi_substruct(),
+                self.get_tile_metadata_name(): self.get_url(AwsConstants.METADATA),
+            }
+        }
+
+    def _get_aux_substruct(self):
+        """Builds a auxiliary data subfolder structure of .SAFE format.
+
+        Note: Old products also have DEM and MSI in aux folder which are not reconstructed here.
+        """
+        aux_safe = {}
 
-        safe[main_folder][AwsConstants.AUX_DATA] = {}
         # Not sure if 2nd condition of the following is correct:
-        if self.data_collection is not DataCollection.SENTINEL2_L1C or self.baseline != "02.04":
+        if self.data_collection is DataCollection.SENTINEL2_L2A or self.baseline != "02.04":
             ecmwft_file = (
                 AwsConstants.ECMWFT
                 if self.data_collection is DataCollection.SENTINEL2_L1C or self.safe_type is EsaSafeType.OLD_TYPE
                 else AwsConstants.AUX_ECMWFT
             )
-            safe[main_folder][AwsConstants.AUX_DATA][self.get_aux_data_name()] = self.get_url(ecmwft_file)
-        # Old products also have DEM and MSI in aux folder
+            aux_safe[self.get_aux_data_name()] = self.get_url(ecmwft_file)
+
+            if self.baseline >= "04.00":
+                camsfo_file = (
+                    AwsConstants.CAMSFO
+                    if self.data_collection is DataCollection.SENTINEL2_L1C
+                    else AwsConstants.AUX_CAMSFO
+                )
```

Comment:
a regular `if else` would be shorter

Reply:
In this case I would actually leave it as it is. Somehow this way seems more readable to me.